### PR TITLE
fix(carousel): flush the slides left when they don't fill the width (#323)

### LINKED ISSUE #323: Slides flushed right instead of left with scrollPerPage
## Bug Report

**Current Behavior**
When there are too few slides to fill the row, with scrollPerPage, the slides are flushed right instead of left.

**Input Code and steps to reproduce**
https://jsfiddle.net/AdminXVII/eywraw8t/473381/

**Expected behavior/code**
In LTR at least, slides should flush left.

**Possible Solution**
Problem seems with maxOffset, which is not constrained to a minimum value of 0.

### PR DESCRIPTION
**Pull request by AdminXVII**
## Description
When there are too few slides to fill the row, with scrollPerPage, the slides were flushed right instead of left.
By bounding maxOffest to a minimum of 0, the slides are aligned on the left-hand-side

## Motivation and Context
Fix #323 

## How Has This Been Tested?
Tested that all features in vue-play worked as before and added a demo to demonstrate behaviour.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [X] I have included a vue-play example (if this is a new feature)

### CODE REVIEW

**Comment by quinnlangille** [comment]
Hey @AdminXVII, thanks for the fix! We'll review sometime today or tomorrow and if all is good merge into the `v0.16.0` working branch :rocket:

File: play/index.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ play("Carousel", module)
       h, containerWidth, [h(Carousel, {}, generateSlideImages(h))]
     )
   )
+  .add("too few per page", h => createContainer(
```
Comment (ashleysimpson) [inline play/index.js]: Do you mind updating this to "Too few per page" just for consistency?

File: src/Carousel.vue [hunk 2]
```diff
@@ -406,9 +406,10 @@ export default {
      * @return {Number}
      */
     maxOffset() {
-      return (
+      return Math.max(
```
Comment (ashleysimpson) [inline src/Carousel.vue]: I see that this change for your example works and also doesn't break the other functionality. The only issue I see is that the pagination breaks but that seems to just be an issue with scrollPerPage.

Comment (quinnlangille) [reply src/Carousel.vue]: Yeah @ashleysimpson and I discussed, this seems like an issue with the `scrollPerPage` api. We'll open an issue to address

File: play/index.js (hunk 1 above)
Comment (AdminXVII) [reply play/index.js]: Done

**Comment by AdminXVII** [comment]
Done

**Comment by quinnlangille** [comment]
Hey @AdminXVII, sorry for the huge delay! Merging into master now and will release as a patch, thanks for the contribution :octocat:

**Pull request merged**